x-pack/plugins/apm/public/services/telemetry/telemetry_client.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AnalyticsServiceSetup } from '@kbn/core-analytics-server';
+import {
+  TelemetryEventTypes,
+  ITelemetryClient,
+  SearchQuerySubmittedParams,
+} from './types';
+
+export class TelemetryClient implements ITelemetryClient {
+  constructor(private analytics: AnalyticsServiceSetup) {}
+
+  public reportSearchQuerySubmitted = ({
```

Comment:
High Level Comment (Open for discussion) - Do we really need this abstraction of a TelemetryClient ?

If i understand correctly what is does is calls the `analytics.reportEvent` function with an `type`

We already have `analytics` service injected in the app, which means we can directly call this from the `UnifiedSearch` component ?

Reason being, we are creating one more service and injecting which can already be called from existing components ?

Reply:
Telemetry team suggests a [best practice](https://docs.elastic.dev/telemetry/collection/event-based-telemetry#best-practices) to abstract the logic. 

> To avoid mistyping event names between registration of event type and reporting the event. We advise adding a small abstraction to share this logic either through creating a class that manages all communications with the analytics service, or simply through the use of constants.



To your questions 
> We already have analytics service injected in the app, which means we can directly call this from the UnifiedSearch component ?

Yes, we can, but what does the future look like when we bring in more event types? `analytics.reportEvent` will be across the whole app making it difficult to know which events are available and reuse

Wrapping all APM event types in one place gives us the benefits of better type safety and the ability to easily reuse these event types.

> Reason being, we are creating one more service and injecting which can already be called from existing components ?

TBH I don't like that part either. But the benefits above outweigh this inmho.  Unless there is a better way, let me know you if you have anything in mind

Reply:
I completely liked the idea of the `tel_service` and `tel_events`. My concern was only around the Client as i felt it was not generic enough. 
Currently the Client has one function `reportSearchQuerySubmitted`

In the future this would increase for example we add 10 more events like `reportEvent1`, `reportEvent2` .....`reportEvent10`. All these methods would do, 

```
this.analytics.reportEvent(EVENT_TYPE, PAYLOAD)
```

But i agree to the Best Practise of not spreading the `analytics.reportEvent` code everywhere in our codebase. Maybe we can revisit this topic once we have more events to report. Thank you for clarifying this 👍🏼